api/src/main/java/org/hawkular/inventory/api/model/MetricType.java

```diff
@@ -157,7 +155,7 @@ private Update() {
 
         public Update(Map<String, Object> properties, MetricUnit unit) {
             super(properties);
-            this.unit = unit;
+            this.unit = unit == null ? MetricUnit.NONE : unit;
```

Comment:
> does the client need to supply all fields or are only the supplied ones updated

the latter, although there is an exception to this for the properties where we allow arbitrarily deep `Map<String, Object>` structure. If the null properties are passed in the update object, nothing is changed, empty hashmap sets it to empty properties; and non-empty hashmap sets (not updates only the keys that are set) it to the passed non-empty hashmap

> should we ever allow null as a valid value for MetricUnits

`null` in context of updating means "don't touch this"

> are there fields like unit that..

agree, it's tricky. I can think of an example where we/feed/whoever discover something and we don't know the units, so we set it to `NONE` and user can manually set the units to enrich the inventory metadata